tor_ocr/main.py

```diff
@@ -76,8 +67,33 @@ def chunks(s, n):
         yield s[start:(start + n)]
 
 
-def run(config):
+def ocr_space_url(url, overlay=False, api_key=os.getenv("OCR_API_KEY", "helloworld"), language='eng'):
+    """ OCR.space API request with remote file.
+        Python3.5 - not tested on 2.7
+        This code is stolen from https://github.com/Zaargh/ocr.space_code_example/blob/master/ocrspace_example.py
+    :param url: Image url.
+    :param overlay: Is OCR.space overlay required in your response.
+                    Defaults to False.
+    :param api_key: OCR.space API key.
+                    Defaults to environment variable "OCR_API_KEY", if it doesn't exist, it will use "helloworld"
+    :param language: Language code to be used in OCR.
+                    List of available language codes can be found on https://ocr.space/OCRAPI
+                    Defaults to 'en'.
+    :return: Result in JSON format.
+    """
 
+    payload = {'url': url,
+               'isOverlayRequired': overlay,
+               'apikey': api_key,
+               'language': language,
+               }
+    r = requests.post('https://api.ocr.space/parse/image',
```

Comment:
Two things: let's rename `r` to `result`, because `r` in the codebase is already reserved for the `reddit` instance. Second, if `r` is a regular requests response, we shouldn't need to decode or parse the response -- `r.json()` should give us the response we're looking for.

Reply:
TIL .json